xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVTBGL.cpp

```diff
@@ -56,38 +51,31 @@ CRenderInfo CRendererVTB::GetRenderInfo()
   return info;
 }
 
-void CRendererVTB::AddVideoPicture(VideoPicture &picture, int index)
+bool CRendererVTB::HandlesVideoBuffer(CVideoBuffer *buffer)
 {
-  YUVBUFFER &buf = m_buffers[index];
-  //CVTBData *vtbdata = (CVTBData*)buf.hwDec;
-//  if (vtbdata->m_vtbbuf)
-//  {
-//    CVBufferRelease(vtbdata->m_vtbbuf);
-//  }
-//  //CVPixelBufferRef cvref = static_cast<CVPixelBufferRef>(picture.hwPic);
-//  vtbdata->m_vtbbuf = cvref;
-//
-//  // retain another reference, this way VideoPlayer and renderer can issue releases.
-//  CVBufferRetain(cvref);
+  VTB::CVideoBufferVTB *vb = dynamic_cast<VTB::CVideoBufferVTB*>(buffer);
+  if (vb)
+    return true;
+
+  return false;
 }
 
 void CRendererVTB::ReleaseBuffer(int idx)
 {
   YUVBUFFER &buf = m_buffers[idx];
-  //if (buf.hwDec)
+  if (buf.videoBuffer)
   {
-    //CVTBData *vtbdata = (CVTBData*)buf.hwDec;
-//    if (vtbdata->m_vtbbuf)
-//    {
-//      CVBufferRelease(vtbdata->m_vtbbuf);
-//      vtbdata->m_vtbbuf = nullptr;
-//    }
-//
-//    if (vtbdata->m_fence && glIsFenceAPPLE(vtbdata->m_fence))
-//    {
-//      glDeleteFencesAPPLE(1, &vtbdata->m_fence);
-//      vtbdata->m_fence = 0;
-//    }
+    VTB::CVideoBufferVTB *vb = dynamic_cast<VTB::CVideoBufferVTB*>(buf.videoBuffer);
+    if (vb)
+    {
+      if (vb->m_fence && glIsFenceAPPLE(vb->m_fence))
+      {
+        glDeleteFencesAPPLE(1, &vb->m_fence);
+        vb->m_fence = 0;
+      }
+    }
+    vb->Release();
```

Comment:
that entire block needs rework. testing that dynamic cast makes no sense in ReleaseBuffer. if it were no vtb type, this would not have been added.